File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.java
import java.util.*;

class Solution {
    public int solution(int[] topping) {
        
        HashMap<Integer, Integer> chulsoo = new HashMap<>();
        HashMap<Integer, Integer> brother = new HashMap<>();
        
        for(int i=0; i<topping.length; i++){
            brother.put(topping[i], brother.getOrDefault(topping[i], 0)+1);
        }
        
        int answer = 0;
        
        for(int i=0; i<topping.length; i++){
            chulsoo.put(topping[i], chulsoo.getOrDefault(topping[i], 0)+1);
            if(brother.get(topping[i]) == 1){
                brother.remove(topping[i]);
            }else{
                brother.put(topping[i], brother.get(topping[i])-1);
            }
            if(chulsoo.size() == brother.size()) answer++;
        }
        
        return answer;
        
    }
}

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
    
    const chulsoo = new Map();
    const brother = new Map();
    
    for(let i=0; i<topping.length; i++){
        if(brother.has(topping[i])){
            brother.set(topping[i], brother.get(topping[i])+1);
        }else{
            brother.set(topping[i], 1);
        }
    }
    
    var answer = 0;
    
    for(let i=0; i<topping.length; i++){
        if(chulsoo.has(topping[i])){
            chulsoo.set(topping[i], chulsoo.get(topping[i])+1);
        }else{
            chulsoo.set(topping[i], 1);
        }
        if(brother.get(topping[i]) == 1){
            brother.delete(topping[i]);
        }else{
            brother.set(topping[i], brother.get(topping[i])-1);
        }
        if(chulsoo.size == brother.size) answer++;
    }
    
    
    return answer;
}